go/pulldown/pulldown.go

```diff
@@ -0,0 +1,176 @@
+/*
+	Copyright 2022 Loophole Labs
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		   http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package pulldown
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/loopholelabs/scalefile"
+)
+
+const POLICY_PULL_ALWAYS = "always"
```

Comment:
This should be its own type. Something like:

```go
type PullPolicy string

const (
  AlwaysPullPolicy PullPolicy = "always"
  IfNotPresentPullPolicy = "if-not-present"
  NeverPullPolicy = "never"
)
```

Reply:
c5699cc899c6eb122a5e4593a173908e2c668c80